### DISCUSSION TITLE
ERROR: Procfile.dev does not exist.

### DISCUSSION THREAD

**Discussion by Gordon2780** [tryzealot/zealot#979] [Q&A] [answered]
### 部署方式 | How to Deploy

源码部署 | Source

### 部署版本 | Version

stable

### 反代服务 | Reverse Proxy

none

### HTTPS 证书类型 | HTTPS(SSL)

自签名证书 Self-signed SSL

### 已阅读文档 | Read the document

- [X] 已阅读 | Yes

### 描述 | Description

Mac Apple M1 
下载 zealot-develop 或 zealot-4.5.3 版本 （不同路径）
按照文档 https://zealot.ews.im/zh-Hans/docs/self-hosted/deployment/source-code
执行 bin/dev 时报错
 ERROR: Procfile.dev does not exist.
 检查了源码目录 Procfile.dev文件存在

cat  Procfile.dev 

web: bin/rails server -p 3000 -b 0.0.0.0
job: bin/sidekiq -C config/sidekiq.yml
js: yarn build --watch
css: yarn build:css --watch

### 问题日志 | Relevant log output

```shell
➜  ~ cd development 
➜  development cd zealot-4.5.3 
➜  zealot-4.5.3 cd bi
cd: no such file or directory: bi
➜  zealot-4.5.3 cd bin
➜  bin dev
zsh: command not found: dev
➜  bin ./dev
Installing foreman...
Fetching foreman-0.87.2.gem
Successfully installed foreman-0.87.2
Parsing documentation for foreman-0.87.2
Installing ri documentation for foreman-0.87.2
Done installing documentation for foreman after 0 seconds
1 gem installed
ERROR: Procfile.dev does not exist.
➜  bin ./dev
ERROR: Procfile.dev does not exist.
➜  bin cd ~ 
➜  ~ cd zealot-develop
➜  zealot-develop cd bin
➜  bin ./dev
ERROR: Procfile.dev does not exist.
➜  bin cd ..
➜  zealot-develop cat Procfile.dev
web: bin/rails server -p 3000 -b 0.0.0.0
job: bin/sidekiq -C config/sidekiq.yml
js: yarn build --watch
css: yarn build:css --watch
➜  zealot-develop
```

**Comment by icyleaf** [comment, marked as answer]
在克隆项目之后所有的命令均在项目根目录执行

**Reply by Gordon2780** [reply]
可以了，谢谢回复。
不过我遇到了另一个问题，想请教一下 我必须安装 esbuild、 sass 和 js的一些依赖吗 如 jquery 
下面是我的终端日志。新手感谢大神指点
16:41:35 js.1   | $ esbuild app/javascript/*.* --bundle --sourcemap --outdir=app/assets/builds --public-path=assets --watch
16:41:35 css.1  | /bin/sh: sass: command not found
16:41:35 css.1  | error Command failed with exit code 127.
16:41:35 css.1  | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
16:41:36 js.1   | ✘ [ERROR] Could not resolve "jquery"
16:41:36 js.1   | 
16:41:36 js.1   |     app/javascript/frameworks.js:1:19:
16:41:36 js.1   |       1 │ import jquery from "jquery"
16:41:36 js.1   |         ╵                    ~~~~~~~~
16:41:36 js.1   | 
16:41:36 js.1   |   You can mark the path "jquery" as external to exclude it from the bundle, which will remove this error.
16:41:36 js.1   | 
16:41:36 js.1   | ✘ [ERROR] Could not resolve "bootstrap"
16:41:36 js.1   | 
16:41:36 js.1   |     app/javascript/frameworks.js:5:7:
16:41:36 js.1   |       5 │ import "bootstrap"
16:41:36 js.1   |         ╵        ~~~~~~~~~~~
16:41:36 js.1   | 
16:41:36 js.1   |   You can mark the path "bootstrap" as external to exclude it from the bundle, which will remove this error.
16:41:36 js.1   | 
16:41:36 js.1   | ✘ [ERROR] Could not resolve "admin-lte"
16:41:36 js.1   | 
16:41:36 js.1   |     app/javascript/frameworks.js:6:7:
16:41:36 js.1   |       6 │ import "admin-lte"
16:41:36 js.1   |         ╵        ~~~~~~~~~~~
16:41:36 js.1   | 
16:41:36 js.1   |   You can mark the path "admin-lte" as external to exclude it from the bundle, which will remove this error.
16:41:36 js.1   | 
16:41:36 js.1   | ✘ [ERROR] Could not resolve "@hotwired/turbo-rails"
16:41:36 js.1   | 
16:41:36 js.1   |     app/javascript/frameworks.js:7:7:
16:41:36 js.1   |       7 │ import "@hotwired/turbo-rails"
16:41:36 js.1   |         ╵        ~~~~~~~~~~~~~~~~~~~~~~~
16:41:36 js.1   | 
16:41:36 js.1   |   You can mark the path "@hotwired/turbo-rails" as external to exclude it from the bundle, which will remove this error.
16:41:36 js.1   | 
16:41:36 js.1   | ✘ [ERROR] Could not resolve "@rails/actioncable"
16:41:36 js.1   | 
16:41:36 js.1   |     app/javascript/channels/index.js:1:29:
16:41:36 js.1   |       1 │ import * as ActionCable from "@rails/actioncable"
16:41:36 js.1   |         ╵                              ~~~~~~~~~~~~~~~~~~~~
16:41:36 js.1   | 
16:41:36 js.1   |   You can mark the path "@rails/actioncable" as external to exclude it from the bundle, which will remove this error.
16:41:36 js.1   | 
16:41:36 js.1   | ✘ [ERROR] Could not resolve "jquery"
16:41:36 js.1   | 
16:41:36 js.1   |     app/javascript/channels/notification_channel.js:3:19:
16:41:36 js.1   |       3 │ import JQuery from "jquery"
16:41:36 js.1   |         ╵                    ~~~~~~~~
16:41:36 js.1   | 
16:41:36 js.1   |   You can mark the path "jquery" as external to exclude it from the bundle, which will remove this error.
16:41:36 js.1   | 
16:41:36 js.1   | 6 of 26 errors shown (disable the message limit with --log-limit=0)
16:41:36 js.1   | [watch] build finished, watching for changes...
16:41:36 css.1  | exited with code 127
16:41:36 system | sending SIGTERM to all processes
16:41:36 job.1  | exited with code 1
16:41:36 js.1   | exited with code 1
16:41:37 web.1  | exited with code 1

**Reply by icyleaf** [reply]
文档漏了安装前端依赖的安装步骤，你在执行下 `yarn install` 后在执行 `bin/dev`

**Reply by Gordon2780** [reply]
谢谢大佬回复。我打开localhost:3000 提示链接不到服务器
终端日志末尾是 👉🏻 exited with code 1 

➜  zealot-develop yarn install
yarn install v1.22.19
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning "admin-lte > bootstrap-switch@3.3.4" has incorrect peer dependency "bootstrap@^3.1.1".
warning "admin-lte > tempusdominus-bootstrap-4@5.39.2" has unmet peer dependency "moment-timezone@^0.5.31".
warning "admin-lte > tempusdominus-bootstrap-4@5.39.2" has unmet peer dependency "tempusdominus-core@5.19.3".
warning "admin-lte > bootstrap-colorpicker > bootstrap@5.2.0" has unmet peer dependency "@popperjs/core@^2.11.5".
[4/4] 🔨  Building fresh packages...
✨  Done in 26.60s.
➜  zealot-develop bin/dev           
16:52:10 web.1  | started with pid 17185
16:52:10 job.1  | started with pid 17186
16:52:10 js.1   | started with pid 17187
16:52:10 css.1  | started with pid 17188
16:52:10 js.1   | yarn run v1.22.19
16:52:10 css.1  | yarn run v1.22.19
16:52:10 css.1  | $ sass app/assets/stylesheets/application.sass.scss:app/assets/builds/application.css --no-source-map --load-path=node_modules --watch
16:52:10 js.1   | $ esbuild app/javascript/*.* --bundle --sourcemap --outdir=app/assets/builds --public-path=assets --watch
16:52:10 job.1  | /opt/homebrew/lib/ruby/gems/3.1.0/gems/bundler-2.3.26/lib/bundler/definition.rb:507:in `materialize': Could not find puma-6.0.0, rails-i18n-7.0.5, pg-1.4.5, redis-4.7.1, lograge-0.12.0, active_model_serializers-0.10.13, graphql-2.0.15, rack-cors-1.1.1, health_check-3.1.0, tiny_appstore_connect-0.1.7, jb-0.8.0, slim-rails-3.5.1, simple_form-5.1.0, friendly_id-5.5.0, kaminari-1.2.2, carrierwave-2.2.3, webp-ffi-0.3.1, faraday-2.7.1, pundit-2.2.0, devise-4.8.1, devise-i18n-1.10.2, omniauth-2.1.0, omniauth-rails_csrf_protection-1.0.1, omniauth-google-oauth2-1.0.1, omniauth-gitlab-3.0.0, omniauth-feishu-0.1.8, net-ldap-0.17.1, pyu-ruby-sasl-0.0.3.3, rubyntlm-0.6.3, openssl-2.2.1, plist-3.6.0, sys-filesystem-1.4.3, vmstat-2.3.1, pghero-3.0.1, sentry-ruby-5.7.0, sentry-rails-5.7.0, sentry-sidekiq-5.7.0, improved_jenkins_client-1.6.7, rqrcode-2.1.2, activejob-status-0.3.0, sidekiq-6.5.8, sidekiq-scheduler-4.0.3, sidekiq-failures-1.0.4, propshaft-0.6.4, jsbundling-rails-1.0.3, cssbundling-rails-1.1.1, stimulus-rails-1.2.1, turbo-rails-1.3.2, app-info-2.8.3, rails-settings-cached-2.8.2, bootsnap-1.15.0, listen-3.7.1, web-console-4.2.0, graphiql-rails-1.8.0, debug-1.7.0, guard-2.18.0, guard-bundler-3.0.0, guard-migrate-2.0.0, guard-rails-0.8.1, guard-sidekiq-0.1.0, terminal-notifier-2.0.0, terminal-notifier-guard-1.7.0, awesome_print-1.9.2, better_errors-2.9.1, binding_of_caller-1.0.0, letter_opener-1.8.1, letter_opener_web-2.0.0, dotenv-rails-2.8.1, rubocop-1.32.0, rubocop-rails-2.15.2, pry-rails-0.3.9, pry-rescue-1.5.2, request_store-1.5.1, case_transform-0.2, jsonapi-renderer-0.2.2, jwt-2.4.1, slim-4.1.0, kaminari-actionview-1.2.2, kaminari-activerecord-1.2.2, kaminari-core-1.2.2, image_processing-1.12.2, ssrf_filter-1.1.1, ffi-1.15.5, ffi-compiler-1.0.1, faraday-net_http-3.0.2, bcrypt-3.1.18, orm_adapter-0.5.0, responders-3.0.1, warden-1.2.9, hashie-5.0.0, rack-protection-2.2.2, oauth2-1.4.10, omniauth-oauth2-1.7.3, json-2.6.2, mixlib-shellout-3.2.7, socksify-1.7.1, terminal-table-3.0.2, chunky_png-1.4.0, rqrcode_core-1.2.0, connection_pool-2.3.0, rufus-scheduler-3.8.2, tilt-2.0.11, CFPropertyList-3.0.5, android_parser-2.5.0, google-protobuf-3.21.7, icns-0.2.0, image_size-3.0.2, ruby-macho-3.0.0, uuidtools-2.2.0, msgpack-1.6.0, rb-fsevent-0.11.1, rb-inotify-0.10.1, bindex-0.8.1, sprockets-rails-3.4.2, formatador-1.1.0, lumberjack-1.2.8, nenv-0.3.0, notiffany-0.1.3, pry-0.13.1, shellany-0.0.1, guard-compat-1.2.1, coderay-1.1.3, debug_inspector-1.1.0, launchy-2.5.0, dotenv-2.8.1, parallel-1.22.1, parser-3.1.2.0, rainbow-3.1.1, regexp_parser-2.5.0, rubocop-ast-1.19.1, ruby-progressbar-1.11.0, unicode-display_width-2.2.0, interception-0.5, mail-2.7.1, temple-0.8.2, mini_magick-4.11.0, ruby-vips-2.1.4, multi_json-1.15.0, multi_xml-0.6.0, chef-utils-17.10.0, fugit-1.7.1, ast-2.4.2, net-protocol-0.1.3, strscan-3.0.4, timeout-0.3.0, et-orbi-1.2.7, raabro-1.4.0 in locally installed gems (Bundler::GemNotFound)
16:52:10 web.1  | /opt/homebrew/lib/ruby/gems/3.1.0/gems/bundler-2.3.26/lib/bundler/definition.rb:507:in `materialize': Could not find puma-6.0.0, rails-i18n-7.0.5, pg-1.4.5, redis-4.7.1, lograge-0.12.0, active_model_serializers-0.10.13, graphql-2.0.15, rack-cors-1.1.1, health_check-3.1.0, tiny_appstore_connect-0.1.7, jb-0.8.0, slim-rails-3.5.1, simple_form-5.1.0, friendly_id-5.5.0, kaminari-1.2.2, carrierwave-2.2.3, webp-ffi-0.3.1, faraday-2.7.1, pundit-2.2.0, devise-4.8.1, devise-i18n-1.10.2, omniauth-2.1.0, omniauth-rails_csrf_protection-1.0.1, omniauth-google-oauth2-1.0.1, omniauth-gitlab-3.0.0, omniauth-feishu-0.1.8, net-ldap-0.17.1, pyu-ruby-sasl-0.0.3.3, rubyntlm-0.6.3, openssl-2.2.1, plist-3.6.0, sys-filesystem-1.4.3, vmstat-2.3.1, pghero-3.0.1, sentry-ruby-5.7.0, sentry-rails-5.7.0, sentry-sidekiq-5.7.0, improved_jenkins_client-1.6.7, rqrcode-2.1.2, activejob-status-0.3.0, sidekiq-6.5.8, sidekiq-scheduler-4.0.3, sidekiq-failures-1.0.4, propshaft-0.6.4, jsbundling-rails-1.0.3, cssbundling-rails-1.1.1, stimulus-rails-1.2.1, turbo-rails-1.3.2, app-info-2.8.3, rails-settings-cached-2.8.2, bootsnap-1.15.0, listen-3.7.1, web-console-4.2.0, graphiql-rails-1.8.0, debug-1.7.0, guard-2.18.0, guard-bundler-3.0.0, guard-migrate-2.0.0, guard-rails-0.8.1, guard-sidekiq-0.1.0, terminal-notifier-2.0.0, terminal-notifier-guard-1.7.0, awesome_print-1.9.2, better_errors-2.9.1, binding_of_caller-1.0.0, letter_opener-1.8.1, letter_opener_web-2.0.0, dotenv-rails-2.8.1, rubocop-1.32.0, rubocop-rails-2.15.2, pry-rails-0.3.9, pry-rescue-1.5.2, request_store-1.5.1, case_transform-0.2, jsonapi-renderer-0.2.2, jwt-2.4.1, slim-4.1.0, kaminari-actionview-1.2.2, kaminari-activerecord-1.2.2, kaminari-core-1.2.2, image_processing-1.12.2, ssrf_filter-1.1.1, ffi-1.15.5, ffi-compiler-1.0.1, faraday-net_http-3.0.2, bcrypt-3.1.18, orm_adapter-0.5.0, responders-3.0.1, warden-1.2.9, hashie-5.0.0, rack-protection-2.2.2, oauth2-1.4.10, omniauth-oauth2-1.7.3, json-2.6.2, mixlib-shellout-3.2.7, socksify-1.7.1, terminal-table-3.0.2, chunky_png-1.4.0, rqrcode_core-1.2.0, connection_pool-2.3.0, rufus-scheduler-3.8.2, tilt-2.0.11, CFPropertyList-3.0.5, android_parser-2.5.0, google-protobuf-3.21.7, icns-0.2.0, image_size-3.0.2, ruby-macho-3.0.0, uuidtools-2.2.0, msgpack-1.6.0, rb-fsevent-0.11.1, rb-inotify-0.10.1, bindex-0.8.1, sprockets-rails-3.4.2, formatador-1.1.0, lumberjack-1.2.8, nenv-0.3.0, notiffany-0.1.3, pry-0.13.1, shellany-0.0.1, guard-compat-1.2.1, coderay-1.1.3, debug_inspector-1.1.0, launchy-2.5.0, dotenv-2.8.1, parallel-1.22.1, parser-3.1.2.0, rainbow-3.1.1, regexp_parser-2.5.0, rubocop-ast-1.19.1, ruby-progressbar-1.11.0, unicode-display_width-2.2.0, interception-0.5, mail-2.7.1, temple-0.8.2, mini_magick-4.11.0, ruby-vips-2.1.4, multi_json-1.15.0, multi_xml-0.6.0, chef-utils-17.10.0, fugit-1.7.1, ast-2.4.2, net-protocol-0.1.3, strscan-3.0.4, timeout-0.3.0, et-orbi-1.2.7, raabro-1.4.0 in locally installed gems (Bundler::GemNotFound)
16:52:10 job.1  | 	from /opt/homebrew/lib/ruby/gems/3.1.0/gems/bundler-2.3.26/lib/bundler/definition.rb:187:in `specs'
16:52:10 web.1  | 	from /opt/homebrew/lib/ruby/gems/3.1.0/gems/bundler-2.3.26/lib/bundler/definition.rb:187:in `specs'
16:52:10 job.1  | 	from /opt/homebrew/lib/ruby/gems/3.1.0/gems/bundler-2.3.26/lib/bundler/definition.rb:244:in `specs_for'
16:52:10 web.1  | 	from /opt/homebrew/lib/ruby/gems/3.1.0/gems/bundler-2.3.26/lib/bundler/definition.rb:244:in `specs_for'
16:52:10 job.1  | 	from /opt/homebrew/lib/ruby/gems/3.1.0/gems/bundler-2.3.26/lib/bundler/runtime.rb:18:in `setup'
16:52:10 web.1  | 	from /opt/homebrew/lib/ruby/gems/3.1.0/gems/bundler-2.3.26/lib/bundler/runtime.rb:18:in `setup'
16:52:10 job.1  | 	from /opt/homebrew/lib/ruby/gems/3.1.0/gems/bundler-2.3.26/lib/bundler.rb:161:in `setup'
16:52:10 web.1  | 	from /opt/homebrew/lib/ruby/gems/3.1.0/gems/bundler-2.3.26/lib/bundler.rb:161:in `setup'
16:52:10 job.1  | 	from /opt/homebrew/lib/ruby/gems/3.1.0/gems/bundler-2.3.26/lib/bundler/setup.rb:20:in `block in <top (required)>'
16:52:10 web.1  | 	from /opt/homebrew/lib/ruby/gems/3.1.0/gems/bundler-2.3.26/lib/bundler/setup.rb:20:in `block in <top (required)>'
16:52:10 job.1  | 	from /opt/homebrew/lib/ruby/gems/3.1.0/gems/bundler-2.3.26/lib/bundler/ui/shell.rb:136:in `with_level'
16:52:10 web.1  | 	from /opt/homebrew/lib/ruby/gems/3.1.0/gems/bundler-2.3.26/lib/bundler/ui/shell.rb:136:in `with_level'
16:52:10 job.1  | 	from /opt/homebrew/lib/ruby/gems/3.1.0/gems/bundler-2.3.26/lib/bundler/ui/shell.rb:88:in `silence'
16:52:10 web.1  | 	from /opt/homebrew/lib/ruby/gems/3.1.0/gems/bundler-2.3.26/lib/bundler/ui/shell.rb:88:in `silence'
16:52:10 job.1  | 	from /opt/homebrew/lib/ruby/gems/3.1.0/gems/bundler-2.3.26/lib/bundler/setup.rb:20:in `<top (required)>'
16:52:10 web.1  | 	from /opt/homebrew/lib/ruby/gems/3.1.0/gems/bundler-2.3.26/lib/bundler/setup.rb:20:in `<top (required)>'
16:52:10 job.1  | 	from <internal:/opt/homebrew/Cellar/ruby/3.1.3/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
16:52:10 web.1  | 	from <internal:/opt/homebrew/Cellar/ruby/3.1.3/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
16:52:10 job.1  | 	from <internal:/opt/homebrew/Cellar/ruby/3.1.3/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
16:52:10 web.1  | 	from <internal:/opt/homebrew/Cellar/ruby/3.1.3/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
16:52:10 job.1  | 	from bin/sidekiq:27:in `<main>'
16:52:10 web.1  | 	from /Users/qtza/zealot-develop/config/boot.rb:5:in `<top (required)>'
16:52:10 web.1  | 	from bin/rails:3:in `require_relative'
16:52:10 web.1  | 	from bin/rails:3:in `<main>'
16:52:11 js.1   | [watch] build finished, watching for changes...
16:52:11 job.1  | exited with code 1
16:52:11 system | sending SIGTERM to all processes
16:52:11 web.1  | exited with code 1
16:52:11 css.1  | exited with code 1
16:52:11 js.1   | exited with code 1

**Reply by icyleaf** [reply]
你执行 bundle install 了么

**Reply by Gordon2780** [reply]
执行了  按照文档的执行顺序执行的

**Reply by icyleaf** [reply]
我看你好像用的是 brew 安装的 Ruby 跟文档用的 Ruby 环境和版本不一样呀

**Reply by Gordon2780** [reply]
对，我用asdf 和rvm分别安装了ruby3.0.0，切换到ruby3.0.0，命令ruby 使用的还是3.1
因为 cocoapods依赖了所以 brew的ruby卸载不了。
我把brew的执行路径和gem的执行路径添加到环境变量 现在用户使用的ruby版本是3.1。

我又重新执行了bundle install 正在安装2.3.4

➜  zealot-4.5.3 bundle install
Bundler 2.3.26 is running, but your lockfile was generated with 2.3.4. Installing Bundler 2.3.4 and restarting using that version.
Fetching gem metadata from https://rubygems.org/..
Fetching bundler 2.3.4

**Reply by icyleaf** [reply]
每个语言环境不同会折腾死人的，不熟悉的话尽量选择文档指导的。

**Reply by Gordon2780** [reply]
你好，源码部署在Mac上面localhost 可以正常使用了。
我想请问下，
如果在docker 部署的话可以用localhost吗？.env配置文件的域名要填写什么
如果在conetos8上面部署源码，安装的环境依赖和Debian (Ubuntu)是一样吗

**Reply by icyleaf** [reply]
本地开发的话试试这个 https://github.com/tryzealot/zealot/pull/983 刚才我又优化了一点点。
CentOS [可以使用 Docker](https://docs.docker.com/engine/install/centos/) 部署的吧
